app/src/server/tasks/evaluateTestSetEntries.task.ts

```diff
@@ -0,0 +1,428 @@
+          } else if (entry.fineTune) {
+            entryIds.push("openpipe:" + entry.fineTune.slug);
+          } else {
+            throw new Error("No fineTune or comparison model found for entry");
+          }
+        } else {
+          entry = result.datasetEvalDatasetEntry.datasetEntry;
+          entryIds.push("the original model");
+        }
+        entries.push(entry);
+      } catch (e) {
+        console.error("error getting entry for result", result, e);
+        await prisma.datasetEvalResult.updateMany({
+          where: {
+            id: {
+              in: [firstResult.id, secondResult.id],
+            },
+          },
+          data: {
+            status: "ERROR",
+            errorMessage: "Error retrieving relevant input for evaluation",
+          },
+        });
+        return;
+      }
+    }
+
+    const [firstEntry, secondEntry] = entries;
+    const [firstEntryId, secondEntryId] = entryIds;
+
+    if (!firstEntry?.output || !secondEntry?.output || !firstEntryId || !secondEntryId) {
+      await prisma.datasetEvalResult.updateMany({
+        where: {
+          id: {
+            in: [firstResult.id, secondResult.id],
+          },
+        },
+        data: {
+          status: "ERROR",
+          errorMessage: "Error preparing for evaluation",
+        },
+      });
+      return;
+    }
+
+    let explanation;
+    let judgement;
+
+    const instructions = firstResult.datasetEvalDatasetEntry.datasetEval.instructions;
+
+    try {
+      const input: ChatCompletionCreateParams = {
+        model: "gpt-4",
+        messages: [
+          {
+            role: "system",
+            content: `You are an intelligent and fair judge of chatbots. Evaluate the following two chatbot responses and choose which one is better in the following way: ${
+              instructions ?? ""
+            }`,
+          },
+          {
+            role: "user",
+            content: formatDatasetEntryInputInstructions(
+              firstResult.datasetEvalDatasetEntry.datasetEntry,
+            ),
+          },
+          {
+            role: "user",
+            content: `This is what Alice said:\n\n${JSON.stringify(firstEntry.output)}`,
+          },
+          {
+            role: "user",
+            content: `This is what Bob said:\n\n${JSON.stringify(secondEntry.output)}`,
+          },
+          {
+            role: "user",
+            content: `Which response is better in the context of the task? Remember to pay attention to the following: ${
+              instructions ?? ""
+            }`,
+          },
+        ],
+        tools: [
+          {
+            type: "function",
+            function: {
+              name: "record_score",
+              parameters: {
+                type: "object",
+                properties: {
+                  explanation: {
+                    type: "string",
+                    description: "An explanation of why you chose the response you did",
```

Comment:
"Explain any factors that might make one response better than the other, and then state which one is higher quality overall. If they're about the same you can just say that."